2009-06-25  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Volumes.pm: Fixed perl syntax error, properly handle existing
		empty volume groups (thanks Brian Kroth and Cajus Pollmeier for debugging
		this).  (closes: #532321)
Index: trunk/lib/setup-storage/Volumes.pm
===================================================================
--- trunk.orig/lib/setup-storage/Volumes.pm
+++ trunk/lib/setup-storage/Volumes.pm	
@@ -305,23 +305,26 @@
   foreach my $vg (get_volume_group_list()) {
     # initialise the hash entry
     $FAI::current_lvm_config{$vg}{physical_volumes} = ();
-    &FAI::push_command( "true", "", "vg_created_$vg" );
+    &FAI::push_command("true", "", "vg_created_$vg");
 
     # store the vg size in MB
     my %vg_info = get_volume_group_information($vg);
-    $FAI::current_lvm_config{$vg}{size} =
-      &FAI::convert_unit( $vg_info{alloc_pe_size} .
-        $vg_info{alloc_pe_size_unit} );
+    if (%vg_info) {
+      $FAI::current_lvm_config{$vg}{size} = $vg_info{alloc_pe_size} .
+        $vg_info{alloc_pe_size_unit};
+    } else {
+      $FAI::current_lvm_config{$vg}{size} = "0M";
+    }
 
-      # store the logical volumes and their sizes
+    # store the logical volumes and their sizes
     my %lv_info = get_logical_volume_information($vg);
     foreach my $lv_name (sort keys %lv_info) {
       my $short_name = $lv_name;
-      $short_name =~ "s{/dev/\Q$vg\E/}{}";
+      $short_name =~ s{/dev/\Q$vg\E/}{};
       $FAI::current_lvm_config{$vg}{volumes}{$short_name}{size} =
         &FAI::convert_unit($lv_info{$lv_name}->{lv_size} .
           $lv_info{$lv_name}->{lv_size_unit});
-      &FAI::push_command( "true", "", "exist_/dev/$vg/$short_name" );
+      &FAI::push_command("true", "", "exist_/dev/$vg/$short_name");
     }
 
     # store the physical volumes
